extensions/azurecore/src/account-provider/azureAccountProvider.ts

```diff
@@ -172,6 +172,11 @@ export class AzureAccountProvider implements azdata.AccountProvider, vscode.Disp
 	}
 
 	refresh(account: AzureAccount): Thenable<AzureAccount | azdata.PromptFailedResult> {
+		return this._refresh(account);
+	}
+
+	private async _refresh(account: AzureAccount): Promise<AzureAccount | azdata.PromptFailedResult> {
```

Comment:
ok I was just trying to follow the same pattern as the prompt function, do you think I should change that one too? Also I would need to change azdata.d.ts file if I modify the original refresh function

Reply:
Oh I see - this is because you added the async. Yeah ok this is fine then. We keep Thenable in the extension APIs to support other types of Promises in extensions so we shouldn't change that.